client/my-sites/theme/main.jsx

```diff
@@ -676,11 +678,11 @@ class ThemeSheet extends Component {
 
 		const plansUrl = siteSlug ? `/plans/${ siteSlug }/?plan=value_bundle` : '/plans';
 
-		const { canonicalUrl, description, name: themeName } = this.props;
+		const { canonicalUrl, description, seoDescription, name: themeName, seoTitle } = this.props;
 		const title =
-			themeName &&
+			( seoTitle || themeName ) &&
 			translate( '%(themeName)s Theme', {
-				args: { themeName },
+				args: { themeName: seoTitle || themeName },
```

Comment:
The `seoTitle` should already be translated from the server, so we'll want to skip passing it to the `%(themeName)s Theme`.